lib/Controller/RoomController.php

```diff
@@ -1161,19 +1161,21 @@ public function getParticipants(bool $includeStatus = false): DataResponse {
 
 			if ($participant->getAttendee()->getActorType() === Attendee::ACTOR_USERS) {
 				$userId = $participant->getAttendee()->getActorId();
-				$user = $this->userManager->get($userId);
-				if (!$user instanceof IUser) {
-					continue;
-				}
-
 				if ($result['lastPing'] > 0 && $result['lastPing'] <= $maxPingAge) {
 					$this->participantService->leaveRoomAsSession($this->room, $participant);
 				}
 
 				if ($this->getAPIVersion() < 3) {
 					$result['userId'] = $participant->getAttendee()->getActorId();
 				}
-				$result['displayName'] = (string) $user->getDisplayName();
+				$result['displayName'] = $participant->getAttendee()->getDisplayName();
+				if (!$result['displayName']) {
```

Comment:
this is true even for the string "0" (corner case though)

Reply:
That's not valid and respected in a lot of other places, so don't care